File: apps/frontend/src/types/version-assignment.ts
export type TicketStatus = 'To Do' | 'In Progress' | 'In Review' | 'Done'
export type FilterPeriod = '15d' | '1m' | '2m' | '3m'

export interface JiraTicket {
  id: string
  summary: string
  epicId: string | null
  epicName: string | null
  status: TicketStatus
  currentVersion: string | null
}

export interface JiraVersion {
  id: string
  name: string
  releaseDate: string
}

export interface EpicGroup {
  epicId: string | null
  epicName: string | null
  tickets: JiraTicket[]
}

File: apps/backend/app/schemas/version_assignment.py
from pydantic import BaseModel


class VersionOption(BaseModel):
    id: str
    name: str
    release_date: str | None


class VersionOptionListResponse(BaseModel):
    versions: list[VersionOption]


class UnversionedTicket(BaseModel):
    id: str
    summary: str
    status: str
    epic_id: str | None
    epic_name: str | None


class UnversionedTicketListResponse(BaseModel):
    tickets: list[UnversionedTicket]


class AssignVersionRequest(BaseModel):
    ticket_ids: list[str]
    version_id: str


class AssignVersionResult(BaseModel):
    succeeded: list[str]
    failed: list[str]
